### ISSUE TITLE
offline support

### ISSUE THREAD

**Issue by dk87** [nextcloud/notes#11]
enable offline support

**Comment by jancborchardt** [comment]
The problem here is that Nextcloud itself as the container is rather server-heavy. Not sure if it is possible to make just the view of one app available offline. I agree it would be pretty cool.

@Henni @korelstar what do you think?

**Comment by korelstar** [comment]
Of course, offline support of the web app would be nice. But since the server has no support for this (instead one can use WebDAV, so there is indeed no need for offline support of the web app), I think this is rather not realizable (with reasonable effort).

But what's the exact use case for an offline support of this web app? Typically this is interesting in mobile scenarios. But here, we most often use mobile devices with other special requirements. Therefore, the Android app (https://github.com/stefan-niedermann/OwnCloud-Notes) was developed with native offline support.

Hence, in my point of view, the real issue is to provide an app for other platforms. Nevertheless, this could also be a HTML5 offline web app. But as long as the Nextcloud server has no API for this, this will have to be a (new) stand-alone app. And here we face again the high effort -- I'm not sure, if this will worth it.

**Comment by Henni** [comment]
I have spoken with @dk87 in person about this. The idea is to allow offline access to the notes app on desktop instead of having to use a different app. This could be solved via Service Workers and might be a great way to learn a bit about this technology.
This issue isn't easy to solve for sure, but it would be a great addition.

**Comment by jancborchardt** [comment]
@Henni yep that would be pretty cool, especially as that also improves the behavior on slow networks or unwanted connection losses. :)

**Comment by codejp3** [comment]
Until the Notes app implements WebDav, I just posted a quick-n-dirty hack that could serve the same purpose.

Basically, any notes that I create offline locally will automatically get added and sync'd with NC when the desktop-client connects to NC and syncs.

It's not a specific offline-mode as requested, but it does allow for the desired ability for making/editing  notes offline and then having them auto-sync whenever you reconnected to NC.
https://github.com/nextcloud/notes/issues/152
